rest-api/openapi/custom.yaml

```diff
@@ -0,0 +1,317 @@
+        '404':
+          description: Project not found
+  /projects/{project_name}/open:
+    post:
+      tags:
+        - Projects
+      summary: Open a project by name
+      operationId: openProjectByName
+      parameters:
+        - name: project_name
+          in: path
+          required: true
+          description: Unique name of the project to be opened
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Project opened successfully
+        '404':
+          description: Project not found
+  /projects/{project_name}/save:
+    post:
+      tags:
+        - Projects
+      summary: Save a project by name
+      operationId: saveProjectByName
+      parameters:
+        - name: project_name
+          in: path
+          required: true
+          description: Unique name of the project to be saved
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Project saved successfully
+        '404':
+          description: Project not found
+  /projects/{project_name}/diagram-editors:
+    get:
+      tags:
+        - Diagrams
+      summary: Get a list of all open diagram editors for a project by name
+      operationId: getDiagramEditorsByProjectName
+      parameters:
+        - name: project_name
+          in: path
+          required: true
+          description: Unique name of the project
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A list of open diagram editors
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/DiagramEditor'
+        '404':
+          description: Project not found
+  /projects/{project_name}/diagrams:
+    get:
+      tags:
+        - Diagrams
+      summary: Get a list of all diagrams for a project by name
+      operationId: getDiagramsByProjectName
+      parameters:
+        - name: project_name
+          in: path
+          required: true
+          description: Unique name of the project
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A list of diagrams
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Diagram'
+        '404':
+          description: Project not found
+components:
+  schemas:
+    Diagram:
+      type: object
+      description: A diagram
+      properties:
+        id:
+          type: string
+          description: Unique identifier of the diagram
+        name:
+          type: string
+          description: Name of the diagram
+        description:
+          type: string
+          description: Description of the diagram
+        documentation:
+          type: string
+          description: Documentation of the diagram
```

Comment:
The `description` field here seems to no longer be used. It apparently got filled in at some point in `getDiagramsByProjectName`, but that code is commented out (and you already tagged it for removal). We should either remove this field or add back the code to fill it in.